miden-lib/asm/sat/tx.masm

```diff
@@ -42,3 +45,60 @@ end
 export.get_output_notes_hash
     exec.epilogue::process_created_notes
 end
+
+#! Increments the number of created notes by one.
+#!
+#! Inputs: []
+#! Outputs: []
+proc.increment_num_created_notes
+    exec.layout::get_num_created_notes
+    add.1
+    exec.layout::set_num_created_notes
+end
+
+#! Creates a new note and returns a pointer to the memory address at which the note is stored.
+#!
+#! Inputs: [ASSET, tag, RECIPIENT]
+#! Outputs: [ptr]
+#!
+#! ASSET is the asset to be included in the note.
+#! tag is the tag to be included in the note.
+#! RECIPIENT is the recipient of the note.
+#! ptr is the pointer to the memory address at which the note is stored.
+export.create_note
+    # validate the asset
+    exec.asset::validate_asset
+    # => [ASSET, tag, RECIPIENT]
+
+    # get the index for the next note to be created
+    exec.layout::get_num_created_notes
+    # => [note_idx, ASSET, tag, RECIPIENT]
+
+    # assert that there is space for a new note
+    dup exec.constants::get_max_num_created_notes lt assert
+    # => [note_idx, ASSET, tag, RECIPIENT]
```

Comment:
If this check happens as a part of `increment_num_create_notes` we could remove it from here.

Reply:
Good catch, let's add it to `incremement_num_created_notes`